### PR TITLE
Fix/dev utils certs

### PR DESCRIPTION
**Pull request by norling**
Updates the dev environment with a dedicated service for certificate handling, as well as updating some versions.

Edit: This ended up with rewriting the way tests are run to make them work with the new way certificates are handled... now you can run tests with:
```
docker compose run tests
```
and 
```
docker compose run integration_tests
```
.

### CODE REVIEW

File: dev_utils/certfixer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+RUN apk add openssl
+
+RUN mkdir -p /certfixer
+
+ADD ./make_certs.sh /certfixer/make_certs.sh
+ADD ./ssl.cnf /certfixer/ssl.cnf
+
+WORKDIR /certfixer
+
+CMD ["/bin/sh", "make_certs.sh"]
```
Comment (jbygdell) [inline dev_utils/certfixer/Dockerfile:12]: This is not needed, you can use the `neicnordic/sda-helm-tests-support` container that already exists.

File: dev_utils/docker-compose.yml [hunk 2]
```diff
@@ -1,66 +1,84 @@
...
-  s3_backend:
-    command: server /data
+  certfixer:
+    build:
+      context: ./certfixer
+    volumes:
+      - pubcert:/pubcert
+      - s3_certs:/s3_certs
+      - mq_certs:/mq_certs
+      - proxy_certs:/proxy_certs
```
Comment (jbygdell) [inline dev_utils/docker-compose.yml:9]:
```suggestion
  certfixer:
    image: neicnordic/sda-helm-tests-support:latest
    command: /bin/sh /certfixer/make_certs.sh
    volumes:
      - ./certfixer:/certfixer
      - pubcert:/pubcert
      - s3_certs:/s3_certs
      - mq_certs:/mq_certs
      - proxy_certs:/proxy_certs
```

File: dev_utils/docker-compose.yml [hunk 3]
```diff
@@ -90,18 +108,64 @@ services:
...
-      - ./certs/proxy.key:/certs/proxy.key
+      - proxy_certs:/certs
       - ./users.csv:/users.csv
       - ./keys:/keys
     ports:
       - "8000:8000"
       - "8001:8001"
 
+  tests:
+    image: golang:${GOLANG_VERSION:-1.18}
```
Comment (jbygdell) [inline dev_utils/docker-compose.yml:119]:
```suggestion
    image: golang:${GOLANG_VERSION:-1.19}
```

File: dev_utils/docker-compose.yml [hunk 4]
```diff
@@ -90,18 +108,64 @@ services:
...
+    container_name: s3proxy-tests
+    profiles:
+      - test
+    command:
+      - "/bin/sh"
+      - "-c"
+      - "cd /app; echo 'Running go ${GOLANG_VERSION:-1.18} tests';
```
Comment (jbygdell) [inline dev_utils/docker-compose.yml:126]:
```suggestion
      - "cd /app; echo 'Running go ${GOLANG_VERSION:-1.19} tests';
```

**Review by jbygdell — CHANGES_REQUESTED** [review]

File: dev_utils/docker-compose.yml (hunk 4 above)
Comment (norling) [reply dev_utils/docker-compose.yml:126]: 1.19 broke something in the linting so I'll update golang versions in a separate PR :) don't want this one to grow out of hand...

**Review by aaperis — APPROVED** [review]
if you like, consider adding the testing compose commands to the dev_utils readme for completeness, but otherwise it looks good to me :-)

**Review by jbygdell — DISMISSED** [review]

**Review by aaperis — APPROVED** [review]
looks great :-)

**Pull request merged**